vmdb/app/models/service_orchestration.rb

```diff
@@ -5,31 +5,38 @@ class ServiceOrchestration < Service
   attr_writer :stack_name
 
   def stack_options
-    @stack_options ||= build_stack_options
+    @stack_options ||= options[:create_options] || build_stack_options_from_dialog
   end
 
   def stack_name
-    @stack_name ||= OptionConverter.get_stack_name(options[:dialog] || {})
+    @stack_name ||= options[:stack_name] || OptionConverter.get_stack_name(options[:dialog] || {})
+  end
+
+  def stack_id
+    @stack_id ||= options[:stack_id]
   end
 
   def orchestration_stack_status
-    orchestration_manager.stack_status(stack_name)
+    return "check_status_failed", "stack has not been deployed" unless stack_id
+
+    orchestration_manager.stack_status(stack_name, stack_id)
   rescue MiqException::MiqOrchestrationStatusError => err
     # naming convention requires status to end with "failed"
     return "check_status_failed", err.message
   end
 
   def deploy_orchestration_stack
-    save_stack_options
-    orchestration_manager.stack_create(stack_name, orchestration_template, stack_options)
+    @stack_id = orchestration_manager.stack_create(stack_name, orchestration_template, stack_options)
     nil  # if no exception
```

Comment:
@bzwei I know this was not changed in the PR, but is there any harm in allowing the stack_id to get passed back?  Maybe we can just remove this line.